sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -416,10 +416,13 @@ static void appendCompileOptionsFromImage(std::string &CompileOpts,
     CompileOpts += isEsimdImage ? "-doubleGRF" : "-ze-opt-large-register-file";
   }
   // Add optimization flags.
-  const char *optLevelStr = getUint32PropAsOptStr(Img, "optLevel");
+  const char *optLevelStr = getUint32PropAsOptStr(Img, "optLevel").c_str();
```

Comment:
Use after free. Returned temporary dies before line 420.

Reply:
Hmm. Sorry. Will need to assign to a local variable. Thanks